### PR TITLE
Feat/pickup modal

### PR DESCRIPTION
**Pull request by jgfidelis**
#### What is the purpose of this pull request?

implement pickup modal and tab

screen shot 2018-12-07 at 14 04 35
screen shot 2018-12-07 at 14 05 00
screen shot 2018-12-07 at 14 09 34

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/Manager.js [hunk 1]
```diff
@@ -62,10 +67,13 @@ class AddressManager extends Component {
       logoUrl,
     }
 
+    const pickupName = path(['pickupPoint', 'friendlyName'], pickupPointQuery)
+    const pickupNameString = pickupName ? pickupName + ' - ' : ''
```
Comment (lbebber) [inline react/Manager.js]: :pick: 

```suggestion
    const pickupNameString = pickupName ? `${pickupName} - ` : ''
```

File: react/components/PickupModalContainer.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  handleSearchAddressChange = async (address) => {
+    const cleanAddress = addValidation(newAddress(address))
+    this.setState({ isSearching: true, searchAddress: cleanAddress })
+    if (cleanAddress.geoCoordinates.value.length < 2) {
+      this.savePickupPoints(null)
+    }
+
+    const [long, lat] = cleanAddress.geoCoordinates.value
```
Comment (lbebber) [inline react/components/PickupModalContainer.js]: achei que fosse o oposto (lat, long)

File: react/components/PickupTab.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+class PickupTab extends Component {
+  state = {
+    isModalOpen: false,
+    askForGeolocation: true,
+    isFetching: false,
+  }
+
+  handleOpenModal = (askForGeolocation) => this.setState({ isModalOpen: true, askForGeolocation })
+
+  onHandleCloseModal = () => this.setState({ isModalOpen: false })
```
Comment (lbebber) [inline react/components/PickupTab.js]: `handleCloseModal`

File: react/components/PickupTab.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+  onHandlePickedSLA = async (pickupPoint) => {
```
Comment (lbebber) [inline react/components/PickupTab.js]: `handlePickedSLA`

File: react/components/PickupTab.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      // Show user that pickup point choice failed?
+    }
+
+    this.setState({ isFetching: false })
+    if (onOrderFormUpdated) {
+      onOrderFormUpdated()
+    }
+  }
+
+  renderAskForModal = () => {
```
Comment (lbebber) [inline react/components/PickupTab.js]: talvez faz um componente separado aqui

File: react/components/PickupTab.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+          >
+            <FormattedMessage id="address-locator.pickup-tab-input-address" />
+          </Button>
+        </div>
+
+      </Fragment>
+    )
+  }
+
+  renderDeliveryPicked = () => {
```
Comment (lbebber) [inline react/components/PickupTab.js]: talvez faz um componente separado aqui

File: react/components/Search.js [hunk 7]
```diff
@@ -190,7 +190,7 @@ class AddressSearch extends Component {
       receiverName: '',
       state: parsedAddressComponents.administrative_area_level_1,
       street: parsedAddressComponents.route,
-      geoCoordinates: latitude && longitude ? [latitude, longitude] : null,
+      geoCoordinates: latitude && longitude ? [longitude, latitude] : null,
```
Comment (lbebber) [inline react/components/Search.js]: 👀

File: react/locales/es.json [hunk 8]
```diff
@@ -23,8 +23,12 @@
     "address-locator.address-manager-description": "Agregar y gestionar sus Direcciones",
     "address-locator.address-manager-button": "Añadir nueva dirección",
     "address-locator.address-manager.logo-title": "Logotipo",
+    "address-locator.pickup-tab": "Retirar",
     "address-locator.graphql-error": "Encontramos problemas. Por favor, inténtelo de nuevo",
     "address-manager.address-list": "Direcciones anteriores",
+    "address-locator.pickup-tab-use-my-geolocation": "UTILIZAR MI UBICACIÓN ACTUAL",
+    "address-locator.pickup-tab-input-address": "BUSCAR DIRECCIÓN MANUALMENTE",
+    "address-locator.pickup-tab-see-all": "VER TODOS LOS PUNTOS DE RETIRADA",
```
Comment (lbebber) [inline react/locales/es.json]: tira o capslock (deixa a formataçao pra ser feita com CSS)

(aqui e nos outros labels em caixa alta tb)

File: react/components/PickupTab.js (hunk 3 above)
Comment (jgfidelis) [reply react/components/PickupTab.js]: foi o lint que mandou colocar `on` na frente...

**Pull request closed without merging**